pkg/kubelet/kubelet_pods.go

```diff
@@ -244,11 +244,12 @@ func (kl *Kubelet) makeBlockVolumes(pod *v1.Pod, container *v1.Container, podVol
 // Kubernetes only mounts on /etc/hosts if:
 // - container is not an infrastructure (pause) container
 // - container is not already mounting on /etc/hosts
+// - pod is a hostNetwork pod
 // Kubernetes will not mount /etc/hosts if:
 // - when the Pod sandbox is being created, its IP is still unknown. Hence, PodIP will not have been set.
 // - Windows pod contains a hostProcess container
```

Comment:
rather than the above addition, rewrite this to:

```suggestion
// - the Pod is on the pod network and PodIP has not yet been set.
// - the Pod is on Windows, and contains a hostProcess container.
```

Reply:
done